### DISCUSSION TITLE
Feature request for training

### DISCUSSION THREAD

**Discussion by iHopist** [brianch/offline-chess-puzzles#13] [Ideas]
HI! Chess coach here. 

It would be nice if the number of moves to solve the puzzle could be set, according to the level of the student. Unfortunately, Lichess only has it until 4 moves, which is not very helpful until the student goes up over a 1200+- in rating. There is only one place where this is possible (chesstempo.com) but those come with another set of troubles and limitations. Is this possible?

Also, it would be great also if the coach could prepare a set with themes and number of games (and any other parameters that are already in the app) and then download/separate them in PGN from the base, as sometimes I would prefer a human pass to re-select the best exercises, and weed out the very obvious ones, the spite-sacrifices than extend unnecessarily the quantity of moves in the puzzle. 

Thanks!

**Comment by brianch** [comment]
Hello! 

I'm not sure I understood the length issue, you meant getting puzzles with more than 4 moves? It seems a bit counterintuitive to me that puzzles with more moves would be better for beginners.

If you want less than 4, Lichess (and this app) have a filter to get puzzles with 1, 2, 3 and 4+ moves (options "One move", "short", "long", and "very long"). But considering it's only possible to filter by one theme at a time, it's a bit useless to be honest, since then you can't filter by an actual tactical theme too. I could add the option to have more than one theme filter to fix this scenario if that would help.

About the second request, I'm working on adding an export to PDF feature (it would have the puzzles' diagrams), I think this would solve this situation, would that work for you?

Thanks a lot for the feedback! :)
